src/network/kafka.rs

```diff
@@ -43,17 +32,18 @@ impl NetworkClient for Client {
     }
   }
 
-  fn network_state(&mut self) -> &mut NetworkState {
-    &mut self.network_state
+  fn session(&mut self) -> &mut Session {
+    &mut self.session
   }
 
-  fn handle_message(&mut self, buffer: &mut ByteBuf) ->ClientErr {
-    let size = buffer.remaining();
+  fn handle_message(&mut self, buffer: &mut BytesMut) -> ClientErr {
+    let size = buffer.remaining_mut();
     let mut res: Vec<u8> = Vec::with_capacity(size);
     unsafe {
       res.set_len(size);
     }
-    buffer.read_slice(&mut res[..]);
+
+    buffer.put_slice(&mut res[..]);
 
     let parsed_request_message = request_message(&res[..]);
```

Comment:
it should be possible to write `&buffer[..]` directly

Reply:
Sounds good (done in c895390)